src/test/java/spoon/test/api/MetamodelTest.java

```diff
@@ -233,49 +237,56 @@ private void assertConceptsEqual(MetamodelConcept expectedConcept, MetamodelConc
 	private void assertPropertiesEqual(MetamodelProperty expectedProperty, MetamodelProperty runtimeProperty) {
 		assertSame(expectedProperty.getRole(), runtimeProperty.getRole());
 		assertEquals(expectedProperty.getName(), runtimeProperty.getName());
-		assertEquals(expectedProperty.getItemValueType().getActualClass(), runtimeProperty.getItemValueType().getActualClass());
-		assertEquals(expectedProperty.getOwnerConcept().getName(), runtimeProperty.getOwnerConcept().getName());
-		assertSame(expectedProperty.getValueContainerType(), runtimeProperty.getValueContainerType());
-		assertEquals(expectedProperty.getValueType(), runtimeProperty.getValueType());
+		assertEquals(expectedProperty.getTypeofItems().getActualClass(), runtimeProperty.getTypeofItems().getActualClass());
+		assertEquals(expectedProperty.getOwner().getName(), runtimeProperty.getOwner().getName());
+		assertSame(expectedProperty.getContainerKind(), runtimeProperty.getContainerKind());
+		assertEquals(expectedProperty.getTypeOfField(), runtimeProperty.getTypeOfField());
 		assertEquals(expectedProperty.isDerived(), runtimeProperty.isDerived());
 		assertEquals(expectedProperty.isUnsettable(), runtimeProperty.isUnsettable());
 	}
 
+	@Test
+	public void testMetamodelCachedInFactory() throws IOException {
+		//contract: Metamodel concepts are accessible
+		Metamodel.getInstance().getConcepts();
+	}
 
 	/*
 	 * this test reports all spoon model elements which are not yet handled by meta model
 	 * actually this is the result
 	 */
 	@Test
 	public void spoonMetaModelTest() {
-		SpoonMetaModel mm = new SpoonMetaModel(new File("./src/main/java"));
+		Factory factory = new FactoryImpl(new DefaultCoreFactory(), new StandardEnvironment());
+		Metamodel mm = Metamodel.getInstance();
 		List<String> problems = new ArrayList<>();
 
 		//detect unused CtRoles
 		Set<CtRole> unhandledRoles = new HashSet<>(Arrays.asList(CtRole.values()));
 
 		mm.getConcepts().forEach(mmConcept -> {
 			mmConcept.getRoleToProperty().forEach((role, mmField) -> {
-				if (mmField.isUnsettable()) {
-					//contract: all unsettable fields are derived too
-					assertTrue("Unsettable field " + mmField + " must be derived too", mmField.isDerived());
```

Comment:
Why did you removed that contract? Is it no more true?

Reply:
it was hard to understand and debug.

now the contract is much easier: if the method or one of its ancestor contains the Derived annotation (resp Unsettable) it is derived (resp unsettable)

the test has been adapted accordingly

Reply:
These lines was the test you required here https://github.com/INRIA/spoon/pull/1904 
I guess the contract "unsettable implies derived" should be still assert by a test. So I suggest to keep these lines